File: api/src/main/java/com/sinergy/chronosync/dto/request/PaginatedClientRequestDTO.java
package com.sinergy.chronosync.dto.request;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

/**
 * Paginated client request DTO.
 */
@Getter
@Setter
@RequiredArgsConstructor
@AllArgsConstructor
public class PaginatedClientRequestDTO extends BasePaginationRequest {
	private String firstName;
	private String lastName;
	private String email;
	private String phone;
	private String uniqueIdentifier;
}

File: gui/src/dtos/requests/PaginatedClientRequestDTO.ts
import { PageableRequest } from '../../models/BaseEntity';

/**
 * Paginated client request DTO for search operations.
 */
export interface PaginatedClientRequestDTO extends PageableRequest {
  firstName?: string;
  lastName?: string;
  email?: string;
  phone?: string;
  uniqueIdentifier?: string;
}
